reactive/jenkins.py

```diff
@@ -177,3 +198,29 @@ def migrate_charm_data():
 def stop():
     service_stop("jenkins")
     status_set("maintenance", "Jenkins stopped")
+
+
+@when("apt.installed.jenkins")
+@when("reset.jenkins.home")
+def reset_jenkins_home():
+    status_set("maintenance", "Configuring JENKINS_HOME")
+
+    home = unitdata.kv().get("storage.dir")
+    if home is None:
+        home = '/var/lib/jenkins'
+    log("Setting Jenkins home to {}".format(home))
+    unitdata.kv().set("jenkins.home", home)
+    # re-import paths to force update
+    importlib.reload(paths)
```

Comment:
The previous line (not counting the comment) changes the jenkins.home unitdata variable which is what many of the variables in paths are based on. Jenkins itself then has its home reset if needed with 'configuration.set_home()'. If the reload of paths isn't done then when any subsequent functions run they don't work correctly.